src/reynir_correct/pattern.py

```diff
@@ -178,15 +178,37 @@ def __init__(self, ann: List[Annotation], sent: Sentence) -> None:
     def get_wordform(self, lemma: str, cat: str, variants: Iterable[str]):
         """ Get correct wordform from BinPackage, 
         given a set of variants """
-
-        # Get rid of argument variants in verbs:
-        realvars = [v for v in variants if not v.isdigit() and v not in SKIPVARS]
+        cases = ["nf", "þf", "þgf", "ef"]
+        if cat == "so":
+            # Get rid of argument variants in verbs:
+            realvars: Iterable[str]  = [v for v in variants if not v.isdigit() and v not in SKIPVARS]
```

Comment:
Probably need to move declaration of realvars to before the if statement:
```
realvars: List[str]
if cat == "so": ...
```

Reply:
Done